examples/runtime-storage/src/lib.rs

```diff
@@ -43,13 +55,49 @@ mod runtime {
         #[ink(constructor)]
         fn new(&mut self) {}
 
-        /// Returns the account balance, read directly from runtime storage
+        /// Returns an account's free balance, read directly from runtime storage
+        ///
+        /// # Key Scheme
+        ///
+        /// A key for the [substrate storage map](https://github.com/paritytech/substrate/blob/dd97b1478b31a4715df7e88a5ebc6664425fb6c6/frame/support/src/storage/generator/map.rs#L28)
+        /// is constructed with:
+        ///
+        /// ```nocompile
+        /// Twox128(module_prefix) ++ Twox128(storage_prefix) ++ Hasher(encode(key))
+        /// ```
+        ///
+        /// For the `System` module's `Account` map, the [hasher implementation](https://github.com/paritytech/substrate/blob/2c87fe171bc341755a43a3b32d67560469f8daac/frame/system/src/lib.rs#L349)
+        /// is `blake2_128_concat`.
         #[ink(message)]
         fn get_balance(&self, account: AccountId) -> Balance {
-            const BALANCE_OF: &[u8] = b"Balances FreeBalance";
-            let key = crypto::blake2_256(&account.to_keyed_vec(BALANCE_OF));
-            let result = self.env().get_runtime_storage::<Balance>(&key[..]);
-            result.unwrap_or_else(|| Ok(0)).unwrap_or_default()
+            let mut key = vec![
+                // Precomputed: Twox128("System")
+                38, 170, 57, 78, 234, 86, 48, 224, 124, 72, 174, 12, 149, 88, 206, 247,
+                // Precomputed: Twox128("Account")
+                185, 157, 136, 14, 198, 129, 121, 156, 12, 243, 14, 136, 134, 55, 29, 169,
+            ];
+
+            let encoded_account = &account.encode();
+
+            let mut blake_128 = Blake2x128::from(vec::Vec::new());
```

Comment:
Not really important but I'd like to keep it as `Vec::new()` without the prepended `vec::` so could you just import it?